src/app/mail/mail-detail/mail-detail.component.html

```diff
@@ -325,15 +325,28 @@ <h5 class="ui-header-subtitle text-dark mb-0">
     <ul class="list-style-none mt-3">
       <ng-container *ngFor="let attachment of mail.attachments">
         <li class="download-attachment-wrapper mr-1" *ngIf="!attachment.is_inline">
-          <a class="download-attachment" [href]="attachment.document" placement="top"
-             [ngbTooltip]="attachment.document | filename" target="_blank" rel="noopener">
+          <app-progress-bar [active]="decryptedAttachments[attachment.id]?.inProgress"></app-progress-bar>
+          <a *ngIf="!attachment.is_encrypted; else encryptedAttachment" class="download-attachment"
+             [href]="attachment.document" placement="top" [ngbTooltip]="attachment.document | filename" target="_blank"
+             rel="noopener">
             <div class="attachment-icon">
               <i class="icon icon-attachment"></i>
             </div>
             <div class="attachment-name">
               {{attachment.document | filename}}
             </div>
           </a>
+          <ng-template #encryptedAttachment>
```

Comment:
we have two templates by same name, maybe we can keep one on this page and use on both places.

Reply:
And we can pass the email (`mail` or `childMail`) as an argument to template

Reply:
Yes you're right